src/qt/bitcoin.cpp

```diff
@@ -282,7 +282,7 @@ void BitcoinApplication::parameterSetup()
 }
 
 void BitcoinApplication::SetPrune(bool prune, bool force) {
-     optionsModel->SetPrune(prune, force);
+    optionsModel->SetPrune(prune ? DEFAULT_PRUNE_SIZE_GB : 0, force);
```

Comment:
In commit "gui: Fix prune settings after intro" (5fd852990cd0e5c94e384fcf6c5fb22891534f4f)

I can't figure out what this commit is doing. What is being fixed? What was the behavior before this commit and what is the behavior after? Description just says "If QSettings are set already, it is required to reset prune size option," which doesn't make sense to me, but I must be missing context that would make it understandable.

Reply:
@ryanofsky
> I can't figure out what this commit is doing. What is being fixed?

Steps to reproduce this bug:

f25c99e735a2abc930752249687d780923e941c8 (before 5fd852990cd0e5c94e384fcf6c5fb22891534f4f)

1. run `bitcoin-qt`, set GUI option as pictured:
![Screenshot from 2019-11-12 22-07-27](https://user-images.githubusercontent.com/32963518/68707088-c1eea880-0599-11ea-939a-6a7ec19658e1.png)
and quit

2. run `bitcoin-qt -choosedatadir`, in intro set check prune:
![Screenshot from 2019-11-12 22-08-04](https://user-images.githubusercontent.com/32963518/68707261-142fc980-059a-11ea-90c7-e1f5c0e6219a.png)

Expected prune target 2 GB = 1907 MiB.
Actually, logged:
```
Prune configured to target 5722 MiB on disk for block and undo files.
```

It seems, I should split first two commits into separated PR, right?

Reply:
See #17696.